DirectProgramming/C++SYCL_FPGA/Tutorials/Features/task_sequences/README.md

```diff
@@ -0,0 +1,249 @@
+
+## Key Implementation Details
+
+This sample demonstrates some key concepts:
+- Basics of instantiating objects of parameterized instance of task_sequence.
+- Using async() function to invoke task functions for the task_sequence object.
+
+This design demonstrates a system structured like this:
+
+![](assets/LoopStructure.svg)
+
+When you write code containing a sequence of loops, each loop in the sequence must wait for the previous loop in the sequence to fully complete. This results in long execution time and poor occupancy of the FPGA datapath:
+
+![](assets/simulation_naive.png)
+
+Observe in the above graphic that it takes many hundreds of clock cycles before the output pipe produces any output.
+Task sequences allow you to schedule loops to run concurrently, so data quickly flows through sequences of loops.
+
+![](assets/simulation_task_sequence.png)
+
+Observe in the above graphic that the output pipe produces results almost immediately.
+
+### Simple Code Example 
+
+To use a task sequence in your design, include the `<sycl/ext/intel/experimental/task_sequence.hpp>` header file in your source code. The `task_sequence` class is a templated class with 3 parameters:
+
+| Template Parameter | Type     | Default Value | Description
+|--------------------|----------|---------------|---
+| Task function      | callable | N/A           | A callable object `f` that defines the asynchronous task to be associated with the `task_sequence`. The callable object `f` must meet the following requirements: <br> • The object `f` must be statically resolvable at compile time, which means it is not a function pointer. <br> • The object `f` must not be an overloaded function. <br> • The return type (`ReturnT`) and argument types (`ArgsT…`) of object f must be resolvable and fixed.
+| Invocation Capacity* | `uint32_t` | 1 | The size of the hardware queue instantiated for `async()` function calls. This parameter value corresponds to the minimum number of outstanding `async()` function calls to be supported. When the outstanding number of `async()` function calls reach this value, further calls may block until the number of outstanding calls is reduced to the `invocation_capacity`. The default value of this parameter is 1.
+| Response Capacity* | `uint32_t` | 1 | The size of the hardware queue instantiated to hold task function results. This parameter value corresponds to the maximum number of outstanding `async()` calls such that all outstanding tasks are guaranteed to make forward progress. Further `async()` calls may block until the number of outstanding calls reduce to the `response_capacity`. The default value of this parameter is 1.
+
+The following example shows how to use task Sequence.  
+
+```c++
+// Task function
+int mult(int a, int b) {
+	 return a * b;
+}
+
+int mult_and_add(int a1, int b1, int a2, int b2) {
+  task_sequence<mult> task1, task2;
+  task1.async(a1, b1);
+  task2.async(a2, b2);
+  return task1.get() + task2.get();
+}
```

Comment:
You should use code from the code sample here; the point of the code sample is to be simple :)